### PR TITLE
Resolve possible macOS application bundle symlink.

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Comment by ccordoba12** [comment]
@mrclary, this looks good to me but let me know when it's ready.

**Comment by mrclary** [comment]
I think it is better to just resolve the link in `start.py`. This means that we don't have to do anything in `spyder-kernels` and also provides robustness for any future uses of `sys.executable`.

If the executable is not in a macOS bundle, then nothing changes. If it is, then it resolves to `../bin/python`.

File: spyder/app/start.py [hunk 1]
```diff
@@ -56,6 +57,12 @@
 from spyder.utils.external import lockfile
 from spyder.py3compat import is_text_string
 
+if sys.platform == "darwin" and sys.executable.endswith("MacOS/python"):
+    # On macOS conda installations, sys.executable may be a symlink
+    # in the application bundle, and therefore should be resolved to
+    # the executable in the environment.
+    sys.executable = os.readlink(sys.executable)
```
Comment (ccordoba12) [inline spyder/app/start.py]:
```suggestion

# On macOS conda installations, sys.executable may be a symlink in the
# application bundle, and therefore should be resolved to the executable in the
# environment.
if sys.platform == "darwin" and sys.executable.endswith("MacOS/python"):
    sys.executable = os.readlink(sys.executable)
```

**Review by ccordoba12 — COMMENTED** [review]
Just a small suggestion for you @mrclary, the rest looks good to me.

**Comment by mrclary** [comment]
Stand by, need to check...

**Comment by mrclary** [comment]
Sorry for the delay. There was a commit that I intended to remove that wasn't removed, and I thought the status bar should also indicate "Internal" when applicable.

Let me know what you think.

**Comment by ccordoba12** [comment]
I'm not so sure about showing `Internal` in the status bar because the idea of that widget is to display the type of environment the console is using, i.e. Conda, Pyenv or Custom (corresponding to any others). `Internal` is not a type of environment.

So, I'd prefer if you revert that change for now (we can discuss about it later).

**Comment by mrclary** [comment]
> So, I'd prefer if you revert that change for now (we can discuss about it later).

Done.

**Pull request merged**